test/functional/test-xhr.js

```diff
@@ -184,6 +184,20 @@ describe('XHR', function() {
           }).to.throw(/Source origin is not allowed/);
         });
 
+        it('should not include __amp_source_origin if disableAmpSourceOrigin ' +
+            'set to true', () => {
+          xhr.fetchJson('/get', {disableAmpSourceOrigin: true});
+          expect(noOrigin(requests[0].url)).to.equal('/get');
+        });
+
+        it('should not include __amp_source_origin if disableAmpSourceOrigin ' +
+            'true & requireAmpResponseSourceOrigin true', () => {
```

Comment:
Again this is about being explicit.  Should somewhat inadvertently specify both but in conflict, the requireAmpResponseSourceOrigin should take precedent (which this verifies).